### DISCUSSION TITLE
General Installation Guide and Comments

### DISCUSSION THREAD

**Discussion by mdw1715** [CriticalAssetManagement/CAMS#127] [General]
This discussion post is intended to allow us to document installation of CAMS and refine the practice over time so that it is easier for less technical users to get started

**Comment by mdw1715** [comment]
### General Installation Guide for Windows to run CAMS locally

### Step 1 - Create a new folder and clone dependencies. After creating a new folder run these commands:
`cd <my new folder>`
`git clone https://github.com/CriticalAssetManagement/CAMS.git`
`git clone https://github.com/CriticalAssetManagement/CAMS-Dashboard.git`
`git clone https://github.com/terminusdb/terminusdb-documents-ui.git`
`git clone https://github.com/terminusdb/terminusdb-react-table.git`
`git clone https://github.com/terminusdb/terminusdb-client-js.git`

### Step 2 - Install NodeJS
- Download NodeJS from here and run the installer https://nodejs.org/en/download/
- After completion, open the Windows Start Menu and type 'Edit System Environment Variables' and open the search results
- In the System Properties window that opens, make sure you are in the 'Advanced' tab and click 'Environment Variables' at the bottom
- In the top pane that appears (User Variables for <your username>) double click on the 'Path' entry
- In the window that opens, add the full path to where you installed NodeJS (usually- C:\Program Files\nodejs)
- Click okay to all open windows and restart your PC

### Step 3 - Installing Dependencies
- Change directories into where you installed each of the cloned repositories with `cd <full path to directory>`
- Run `npm install` in each

### Step 4 - Run
`npm run start`

The app should be visible if you type http://localhost:3036/ into your web browser

**Comment by mdw1715** [comment]
Currently required that TerminusDB-documents-ui is installed from the DEV branch -
`git install https://github.com/terminusdb/terminusdb-documents-ui.git@dev`

Also crucial to make sure these paths actually point at the directories you downloaded the dependencies to 
![image](https://user-images.githubusercontent.com/35280953/154319398-60aa07df-e9ef-4766-997e-7adfdda9ddd5.png)

**Comment by mdw1715** [comment]
Also requires: https://github.com/terminusdb/react-jsonschema-form

`git clone https://github.com/terminusdb/react-jsonschema-form.git`

**Comment by rrooij** [comment]
I modified the `package.json` in such a way that all the local clones are not needed anymore. You should only need CAMS-dashboard. :smile:

**Reply by bmaddy** [reply]
Thanks for this huge improvement. I'm still running into a bunch of errors like the following:
```
...
web_1  | ERROR in ./src/components/Form.js 2:0-66
web_1  | Module not found: Error: Can't resolve '@terminusdb/terminusdb-documents-ui' in '/app/src/components'
web_1  |  @ ./src/components/Display.js 5:0-30 37:48-52 56:38-42 77:38-42
web_1  |  @ ./src/pages/AssetForm.js 12:0-101 101:38-54 118:115-127 125:60-72 135:38-50 145:48-62
web_1  |  @ ./src/App.js 4:0-46
web_1  |  @ ./src/index.js 3:0-28 42:36-39
web_1  | 
web_1  | ERROR in ./src/components/Table.js 2:0-63
web_1  | Module not found: Error: Can't resolve '@terminusdb/terminusdb-react-table' in '/app/src/components'
web_1  |  @ ./src/pages/Reports.js 18:0-44 88:38-43
web_1  |  @ ./src/App.js 5:0-42 45:46-53
web_1  |  @ ./src/index.js 3:0-28 42:36-39
web_1  | 
web_1  | ERROR in ./src/components/Views.js 3:23-63
web_1  | Module not found: Error: Can't resolve '@terminusdb/terminusdb-client' in '/app/src/components'
web_1  |  @ ./src/pages/AssetForm.js 8:0-53 104:12-26
web_1  |  @ ./src/App.js 4:0-46
web_1  |  @ ./src/index.js 3:0-28 42:36-39
...
```
The libraries appear to exist:
```bash
$ docker-compose up -d
$ docker-compose exec web ls node_modules/@terminusdb
rjsf-core			     terminusdb-documents-ui
terminusdb-access-control-component  terminusdb-react-table
terminusdb-client
```
Any idea what's wrong? If it's not obvious, I haven't done js in years....
